### PR TITLE
Add new package: AutoDock-GPU.

### PR DESCRIPTION
**Pull request by RemiLacroix-IDRIS**
The project currently only has a develop branch and no versioned releases so only provide a git-based "develop" version.

### CODE REVIEW

File: var/spack/repos/builtin/packages/autodock-gpu/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    depends_on('cuda')
+
+    @property
+    def build_targets(self):
+        spec = self.spec
+        return [
+            'DEVICE={0}'.format(spec.variants['device'].value.upper()),
+            'GPU_INCLUDE_PATH={0}'.format(spec['cuda'].prefix.include),
+            'GPU_LIBRARY_PATH={0}'.format(spec['cuda'].prefix.lib64),
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/autodock-gpu/package.py]: Is it always guaranteed to be `lib64` on every system? It's probably better to use `spec['cuda'].libs.directories[0]`

Comment (RemiLacroix-IDRIS) [reply var/spack/repos/builtin/packages/autodock-gpu/package.py]: I wasn't completely sure myself, I think your suggestion is better.

**Review by adamjstewart — APPROVED** [review]

**Comment by RemiLacroix-IDRIS** [comment]
I was hoping an official release but I guess we can start with just a `develop` version.

**Comment by adamjstewart** [comment]
One of the tests failed so I restarted them. Let me know when they finish and I can merge.

**Comment by RemiLacroix-IDRIS** [comment]
@adamjstewart Should be ok now!

**Pull request merged**